plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
On Tue, May 17, 2016 at 04:03:58PM -0700, Michael Grosser wrote:
> >      def deployment_labels
> > -      # Deployment labels should not include the ids of the release, role or deploy groups
> > -      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
> > +      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
> 
> yep, just confirmed that too ... updating it to only use project and role ... does that work for you too ?

Hmm, not sure that will be enough. Maybe it is because staging/production, if
they are in the same cluster, will be in different namespaces. But with the
environment (deploy_group I think?) for sure it is unique